### PR TITLE
[networkx] Riase `NetworkXError` when gs graph is multigraph

### LINKED ISSUE #796: Support holding immutable graph in nx.Graph and `copy-on-write` to dynamic graph
**Is your feature request related to a problem? Please describe.**
To improve the compatibility of networkx module, we should support holding immutable graph in nx.Graph.
- If a nx.Graph holding a immutable graph, it can run  all the APIs which not modify graph.
- If the calling Api is a modification operation, convert to dynamic graph. （copy-on-write)

**Describe the solution you'd like**
- dynamic fragment support vertex label, preserve labels in to_nx_graph(). #804
- use pair (label, id) as alternative node input and output. (current id means it has a default label) #806 
- revise the `Reporter` and `Viewer` to support `arrow_fragment` and label. #805 
- revise edges iterator to support label property graph on app. #807 
- raise error if the oid is pair. #803 
- check multi-edge when loading graph #810 
- **Not breaking the original NetworkX tutorial and tests.**

**A tutorial example**
- tutorial that nx.Graph holding immutable graph and copy-on-write
```python
import graphscope.nx as nx
from graphscope.framework.loader import Loader
graph = graphscope.g(directed=False)
graph = graph.add_vertices(Loader("/home/jovyan/datasets/ldbc_sample/person_0_0.csv", delimiter="|"), "person")
graph = graph.add_edges(Loader("/home/jovyan/datasets/ldbc_sample/person_knows_person_0_0.csv", delimiter="|"),
                        "knows",
                        src_label="person",
                        dst_label="person",
              )
G = nx.Graph(graph, default_label='person')   # holding immutable graph, use `person` as default_label
list(G.nodes)   # if default_label nodes return oid, other label nodes return (label, oid)
list(G.edges)  # if default_label nodes return oid, other label nodes return (label, oid)
G[('person', 0)]  # euqal to G[0]
G.edges[(0, 1)]  # equal to G.edges[('person', 0), ('person, 1)]
G.adj[0][1]   # equal to G.adj[('person', 0)[('person', 1)]
nx.single_source_dijkstra_path_length(G, ('person', 0)) # default label node only need oid, other label need 
nx.single_source_dijkstra_path_length(G, 0) # default label node only need oid, other label need 

G.add_edges_from([(2, 3), (('person', 1), ('person', 2))])  # modify op: convert to dynamic graph and holding dynamic graph.
list(G.nodes) # holding dynamic graph
list(G.edges) # holding dynamic graph
nx.single_source_dijkstra_path_length(G, 0) # running on dynamic graph
```

- tutorial that graphscope graph support NetowrkX api
```python
import graphscope.nx as nx
from graphscope.framework.loader import Loader
graph = graphscope.g(directed=False)
graph = graph.add_vertices(Loader("/home/jovyan/datasets/ldbc_sample/person_0_0.csv", delimiter="|"), "person")
graph = graph.add_edges(Loader("/home/jovyan/datasets/ldbc_sample/person_knows_person_0_0.csv", delimiter="|"),
                        "knows",
                        src_label="person",
                        dst_label="person",
              )
list(graph.nodes)
list(graph.edges)
simple_graph = graph.project(vertices={"person": []}, edges={"knows": ["dist"]})
nx.single_source_dijkstra_path_length(simple_graph, 0) # running on immutable graph

G = nx.Graph(graph) # convert to dynamic graph
G.add_edges_from([(2, 3)])
list(G.nodes)
list(G.edges)
list(G)
nx.single_source_dijkstra_path_length(G, 0)
```

**Pros and cons**
- 优点：
  - 在app计算时可以直接在immutable graph上进行计算 （两个tutorial都支持）
  - tutorial上显式地只有一个nx.Graph，与NetworkX tutorial一致；
- 缺点：
  - ~~基本nx.Graph的涉及点和边的接口都要加入一个label参数，默认值为`_` （包括修改图的接口）~~
  - ~~无论nx.Graph holding的是immutable graph 还是 dynamic graph，都要在调用app前显式地project一次~~
  - ~~一些语法糖接口在上述例子中可能无法使用，如 `for-in` 语法糖，默认取得是'_'  label的点，示例中并不存在; 类似的还有 `G[0]`, `0 in G`,  property method `nodes`, `edges`~~

### LINKED ISSUE #810: [NetworkX] add a step to check multi edges in property graph loading
- check multi edges when loading property graph.
- add a `is_multigraph` to graph meta.

### PR DESCRIPTION
**Pull request by acezen**
## What do these changes do?

Add parallel edge check when init nx graph with  gs graph 

## Related issue number

Fixes #810
Fixes #796 

vineyard check parallel edge issue: https://github.com/v6d-io/v6d/pull/563

### CODE REVIEW

File: analytical_engine/core/grape_instance.cc [hunk 1]
```diff
@@ -908,6 +908,7 @@ bl::result<std::shared_ptr<DispatchResult>> GrapeInstance::OnReceive(
   switch (cmd.type) {
   case rpc::CREATE_GRAPH: {
     BOOST_LEAF_AUTO(graph_def, loadGraph(params));
+    syncGraphDef(graph_def);
```
Comment (siyuan0322) [inline analytical_engine/core/grape_instance.cc:911]: This is the ad-hoc way to sync the result for graph only. I think we have discussed doing this by setting the `AggregatePolicy` of the result, and do this in the aggregate function, which is defined in `graphscope_service.cc`. And the new policy can be named `kMerge`

File: python/graphscope/nx/classes/graph.py [hunk 2]
```diff
@@ -2143,6 +2143,10 @@ def _init_with_arrow_property_graph(self, arrow_property_graph):
                 "Try to init with another direction type's arrow_property graph."
                 + "Graphs must have the same direction type."
             )
+        if arrow_property_graph._is_multigraph:
+            raise NetworkXError(
+                "Graph contain parallel edge, can not convert to networkx graph."
```
Comment (siyuan0322) [inline python/graphscope/nx/classes/graph.py:2148]: wording:  contain -> contains,    convert to -> cannot be converted to,

File: python/graphscope/nx/tests/test_copy_on_write.py [hunk 3]
```diff
@@ -262,6 +289,10 @@ def setup_method(self):
         #     ).values
         # )
 
+    def test_error_with_multigraph(self):
+        with pytest.raises(NetworkXError):
```
Comment (siyuan0322) [inline python/graphscope/nx/tests/test_copy_on_write.py:293]: Better to also specify the `matches` parameter, for instance, match the `parallel edge`

File: analytical_engine/core/server/graphscope_service.cc [hunk 4]
```diff
@@ -82,7 +69,30 @@ ::grpc::Status GraphScopeService::RunStep(::grpc::ServerContext* context,
       return Status(StatusCode::INTERNAL, error_msgs);
     }
 
-    // Second pass: aggregate result according to the policy
+    // Second pass: aggregate graph def according to the policy
+    switch (policy) {
+    case DispatchResult::AggregatePolicy::kMerge: {
```
Comment (siyuan0322) [inline analytical_engine/core/server/graphscope_service.cc:74]: Better to leave some comments about why the graph_def is special and why need to merge, for others can understand this part more quickly.

**Review by siyuan0322 — APPROVED** [review]
LGTM

**Review by yecol — APPROVED** [review]

**Pull request merged**